plugins/bkpr/recorder.c

```diff
@@ -1964,22 +1965,22 @@ void maybe_closeout_external_deposits(struct db *db,
 						    " blockheight = ?"
 						    " WHERE id = ?"));
 
-		db_bind_int(update_stmt, ev->blockheight);
+		db_bind_int(update_stmt, blockheight);
 		db_bind_u64(update_stmt, id);
 		db_exec_prepared_v2(take(update_stmt));
 	}
 
 	tal_free(stmt);
 }
 
-bool log_chain_event(struct db *db,
+bool log_chain_event(const tal_t *ctx, struct db *db,
 		     const struct account *acct,
 		     struct chain_event *e)
 {
 	struct db_stmt *stmt;
 
 	/* We're responsible for de-duping chain events! */
-	if (find_chain_event(e, db, acct,
+	if (find_chain_event(ctx, db, acct,
```

Comment:
This should actually be tmpctx!

Reply:
And ctx arg is unnecessary...

Reply:
wow yeah, lots of gymnastics on this one. fixing..